core/trino-spi/src/main/java/io/trino/spi/block/AbstractArrayBlock.java

```diff
@@ -103,6 +106,12 @@ public Block getRegion(int position, int length)
                 getRawElementBlock());
     }
 
+    @Override
+    public OptionalInt fixedSizeInBytesPerPosition()
```

Comment:
nit: maybe just `getSizeInBytesPerPosition`?

Reply:
I think the "fixed" adds extra information here and makes it more clear (to me) that this method is only for blocks where the size in bytes is not dependent on the value at any given position, but that's just my opinion- if you feel strongly then I'm willing to change it.

Reply:
The semantics is was kinda natural for me even with shorter name based on the fact that single number is returned and is guarded by `Optional` to handle case when we cannot compute it. But I do not feel strongly. it may stay.